From e921fcd2cfe4d6523cd26e2801b6da105fb90dee Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:39 -0500
Subject: [drm] i915/hdmi: fix hdmi audio state readout

Message-id: <1418852380-13061-29-git-send-email-rclark@redhat.com>
Patchwork-id: 102159
O-Subject: [RHEL7 drm 28/89] drm/i915/hdmi: fix hdmi audio state readout
Bugzilla: 1153301
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1153301
Upstream: since v3.16.4

commit 90eebce35adabb7e0a220c36c71a2ebd00dc5e33

Author:     Jani Nikula <jani.nikula@intel.com>
AuthorDate: Wed Sep 17 15:34:58 2014 +0300
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:16:25 2014 -0500

    drm/i915/hdmi: fix hdmi audio state readout

    commit c84db77010877da6c5da119868ed54c43d59e726 upstream.

    Check the correct bit for audio. Seems like a copy-paste error from the
    start:

    commit 9ed109a7b445e3f073d8ea72f888ec80c0532465
    Author: Daniel Vetter <daniel.vetter@ffwll.ch>
    Date:   Thu Apr 24 23:54:52 2014 +0200

        drm/i915: Track has_audio in the pipe config

    Reported-by: Martin Andersen <martin.x.andersen@gmail.com>
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=82756
    Cc: Daniel Vetter <daniel.vetter@ffwll.ch>
    Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
    Signed-off-by: Jani Nikula <jani.nikula@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index eee2bbe..0573664 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -728,7 +728,7 @@ static void intel_hdmi_get_config(struct intel_encoder *encoder,
  if (tmp & HDMI_MODE_SELECT_HDMI)
   pipe_config->has_hdmi_sink = true;
 
- if (tmp & HDMI_MODE_SELECT_HDMI)
+ if (tmp & SDVO_AUDIO_ENABLE)
   pipe_config->has_audio = true;
 
  pipe_config->adjusted_mode.flags |= flags;
-- 
1.7.1